apps/examples/power/power_main.c

```diff
@@ -33,54 +33,51 @@
 /****************************************************************************
  * Pre-processor Definitions
  ****************************************************************************/
-#define PM_LOCK_PATH						"/proc/power/domains/0/pm_lock"
-#define PM_UNLOCK_PATH					"/proc/power/domains/0/pm_unlock"
-#define PM_SLEEP_PATH						"/proc/power/domains/0/enter_sleep"
+#define POWER_THREAD_SIZE				2048
+#define POWER_THREAD_PRIORITY				255
+#define PM_DAEMON_INTERVAL				5
 
-#define POWER_THEAD_SIZE				2048
-#define POWER_THEAD_PRIORITY		255
-#define PM_DAEMON_INTERVAL			15
-#define PM_LOCK(x)								power_set_int(PM_LOCK_PATH, x)
-#define PM_UNLOCK(x)							power_set_int(PM_UNLOCK_PATH, x)
+#define PM_WAKEUP_TIMER_DURATION			5
 
 /****************************************************************************
  * Private Functions
  ****************************************************************************/
-static int power_write_buf(char *file, char *buf)
+#ifdef CONFIG_ARCH_CORTEXA32
+static int power_write_buf(char *file, bool use_timer)
+#else
+static int power_write_buf(char *file)
+#endif
 {
 	int fd;
-	int r;
 	int ret = 0;
 
 	fd = open(file, O_WRONLY);
 	if (fd == -1)
 		return -ENOENT;
 
-	r = write(fd, buf, strlen(buf));
-	if (r < 0)
-		ret = -EIO;
+#ifdef CONFIG_ARCH_CORTEXA32
+	if (strncmp(file, PM_UNLOCK_PATH, 31) == 0 && use_timer) {
+		// state unlock call to pm procfs can set timer for waking up from sleep, if required
+		ret = write(fd, NULL, PM_WAKEUP_TIMER_DURATION);
+		if (ret < 0)
+			ret = -EIO;
+	} else
+#endif
+	{
+		// Dummy write call to procfs, no data actually needs to be written
+		ret = write(fd, NULL, 0);
+		if (ret < 0)
+			ret = -EIO;
+	}
 
 	close(fd);
 
 	return ret;
 }
 
-int power_set_int(char *fname, int val)
-{
-	char buf[8];
-	int ret = 0;
-
-	snprintf(buf, sizeof(buf), "%d", val);
-
-	if (power_write_buf(fname, buf) != 0)
-		ret = -EIO;
-
-	return ret;
-}
-
 static pthread_addr_t power_daemon()
 {
-	char *state = "sleep";
+	int ret = 0;
```

Comment:
We should initialize the value as int ret = -1? Setting to 0 assumes it is correct initially.

Reply:
Does not matter in this case as it will always be changed by the very first function call of PM_UNLOCK